frontend/src/pages/connectionTypes/create/CreateConnectionTypePage.tsx

```diff
@@ -0,0 +1,147 @@
+import * as React from 'react';
+import {
+  Button,
+  Checkbox,
+  Form,
+  FormGroup,
+  FormSection,
+  PageSection,
+} from '@patternfly/react-core';
+import { useNavigate } from 'react-router';
+import { OpenDrawerRightIcon } from '@patternfly/react-icons';
+import { createConnectionType } from '~/services/connectionTypesService';
+import NameDescriptionField from '~/concepts/k8s/NameDescriptionField';
+import { ConnectionTypeConfigMapObj, ConnectionTypeField } from '~/concepts/connectionTypes/types';
+import ConnectionTypePreviewDrawer from '~/concepts/connectionTypes/ConnectionTypePreviewDrawer';
+import {
+  createConnectionTypeObj,
+  extractConnectionTypeFromMap,
+} from '~/concepts/connectionTypes/createConnectionTypeUtils';
+import { translateDisplayNameForK8s } from '~/concepts/k8s/utils';
+import ApplicationsPage from '~/pages/ApplicationsPage';
+import { NameDescType } from '~/pages/projects/types';
+import { CreateConnectionTypeFooter } from './CreateConnectionTypeFooter';
+import { CreateConnectionTypeFieldsTable } from './CreateConnectionTypeFieldsTable';
+import { CreateConnectionTypeBreadcrumbs } from './CreateConnectionTypeBreadcrumbs';
+
+type CreateConnectionTypePageProps = {
+  prefill?: ConnectionTypeConfigMapObj;
+};
+
+export const CreateConnectionTypePage: React.FC<CreateConnectionTypePageProps> = ({ prefill }) => {
+  const navigate = useNavigate();
+
+  const [isDrawerExpanded, setIsDrawerExpanded] = React.useState(false);
+
+  const [prefillNameDesc, prefillEnabled, prefillFields] = extractConnectionTypeFromMap(prefill);
+
+  const [connectionNameDesc, setConnectionNameDesc] = React.useState<NameDescType>(prefillNameDesc);
+  const [connectionEnabled, setConnectionEnabled] = React.useState<boolean>(prefillEnabled);
+  const [connectionFields] = React.useState<ConnectionTypeField[]>(prefillFields);
+
+  const previewConnectionTypeObj = React.useMemo(
+    () =>
+      createConnectionTypeObj(
+        {
+          k8sName: translateDisplayNameForK8s(connectionNameDesc.name),
+          displayName: connectionNameDesc.name,
+          description: connectionNameDesc.description,
+          enabled: connectionEnabled,
+          username: '',
+        },
+        connectionFields,
+      ),
+    [connectionNameDesc, connectionEnabled, connectionFields],
+  );
+
+  const [isCreateLoading, setIsCreateLoading] = React.useState(false);
+  const [createError, setCreateError] = React.useState<string>();
+  const isValid = React.useMemo(() => Boolean(connectionNameDesc.name), [connectionNameDesc.name]);
+
+  const onSave = async () => {
+    if (isValid) {
+      setIsCreateLoading(true);
+      const response = await createConnectionType(previewConnectionTypeObj);
+      if (response.error) {
+        setCreateError(response.error);
```

Comment:
You should clear the error before submitting:
```suggestion
      setIsCreateLoading(true);
      setCreateError(null);
      const response = await createConnectionType(previewConnectionTypeObj);
      if (response.error) {
        setCreateError(response.error);
```